modules/streams/arrow-streams/src/test/kotlin/arrow/streams/internal/FreeCTest.kt

```diff
@@ -0,0 +1,180 @@
+package arrow.streams.internal
+
+import arrow.Kind
+import arrow.core.*
+import arrow.test.UnitSpec
+import arrow.test.generators.genFunctionAToB
+import arrow.test.generators.genThrowable
+import arrow.test.laws.*
+import arrow.typeclasses.Eq
+import io.kotlintest.KTestJUnitRunner
+import io.kotlintest.properties.Gen
+import io.kotlintest.properties.forAll
+import org.junit.runner.RunWith
+
+@RunWith(KTestJUnitRunner::class)
+class FreeCTest : UnitSpec() {
+
+  init {
+
+    testLaws(FunctorLaws.laws(
+      FF = FreeC.functor<ForTry>(),
+      EQ = Eq { a, b -> a.run(Try.monadError()) == b.run(Try.monadError()) },
+      f = { FreeC.pure(it) }
+    ))
+
+    testLaws(ApplicativeLaws.laws(
+      A = FreeC.applicative<ForTry>(),
+      EQ = Eq { a, b -> a.run(Try.monadError()) == b.run(Try.monadError()) }
+    ))
+
+    testLaws(MonadLaws.laws(
+      M = FreeC.monad<ForTry>(),
+      EQ = Eq { a, b -> a.run(Try.monadError()) == b.run(Try.monadError()) }
+    ))
+
+    testLaws(MonadErrorLaws.laws(
+      M = FreeC.monadError<ForTry>(),
+      EQ = Eq { a, b -> a.run(Try.monadError()) == b.run(Try.monadError()) },
+      EQERR = Eq { a, b -> a.run(Try.monadError()) == b.run(Try.monadError()) },
+      EQ_EITHER = Eq { a, b -> a.run(Try.monadError()) == b.run(Try.monadError()) }
+    ))
+
+    testLaws(ApplicativeErrorLaws.laws(
+      AE = FreeC.applicativeError<ForTry>(),
+      EQ = Eq { a, b -> a.run(Try.monadError()) == b.run(Try.monadError()) },
+      EQERR = Eq { a, b -> a.run(Try.monadError()) == b.run(Try.monadError()) },
+      EQ_EITHER = Eq { a, b -> a.run(Try.monadError()) == b.run(Try.monadError()) }
+    ))
+
+    testLaws(MonadDeferLaws.laws(
```

Comment:
I thought the laws where encoded such as if you just test the `MonadDeferLaws` it will include all others in its hierarchy.

Reply:
Yes, they're. This is left over from when I was debugging combinators in smaller sets at a time :D